src/components/profile/ProfileSetPassword.jsx

```diff
@@ -0,0 +1,128 @@
+import { useRequest } from "ahooks";
+import KeyIcon from "@mui/icons-material/Key";
+import { useForm } from "react-hook-form";
+import { useAuth } from "@/models";
+import { useNavigate } from "react-router-dom";
+
+export default function ProfileSetPassword() {
+  const auth = useAuth();
+  const navigate = useNavigate();
+  const { register, handleSubmit } = useForm();
+
+  const GetErrorMessage = (error) => {
+    if (error.code === "password_not_match") {
+      return "Passwords do not match";
+    } else if (error.code === "auth/too-many-requests") {
+      return "Too many trails, try again later";
+    } else if (error.code === "auth/wrong-password") {
+      return "Wrong password";
+    } else {
+      return "Unknown error";
+    }
+  };
+
+  const {
+    run: modifiedContent,
+    error: modificationError,
+    loading: modificationLoading,
+  } = useRequest(
+    async (data) => {
+      if (data.newPassword !== data.checkPassword) {
+        let error = new Error("password_not_match");
+        error.code = "password_not_match";
+        throw error;
+      } else {
+        return auth.updatePassword(data);
+      }
+    },
+    {
+      manual: true,
+      onSuccess: async () => {
+        navigate("../account", { replace: true });
+      },
+      onError: (error) => {
+        console.log(error);
+      },
+    }
+  );
+
+  const handleCancel = () => {
+    navigate("../account", { replace: true });
+  };
+
+  return (
+    <div className="w-full">
+      <form onSubmit={handleSubmit(modifiedContent)}>
+        <div className="w-full flex flex-col items-center">
+          <ul className="menu bg-base-100 px-12 w-full rounded-box">
+            <li>
+              <div className="flex justify-between">
+                <div className="flex justify-start gap-4">
+                  <KeyIcon />
+                </div>
+                <div className="flex w-full justify-end">
+                  <input
+                    {...register("oldPassword", { required: true })}
+                    type="text"
+                    placeholder="old password"
+                    className={
+                      "input w-full input-bordered" +
+                      (modificationError ? " input-error" : "")
```

Comment:
what an error input box means: the input does not meet the format, this happens before the submission.

this should not be an error indicator for an error response